### ISSUE TITLE
Using ppx_expect does not work with reason

### ISSUE THREAD

**Issue by tcoopman** [facebook/reason#2280]
I'm trying to use `ppx_expect`. I transformed this ocaml sample:

```ocaml
let%expect_test _ =
  print_endline "Hello, world!";
  [%expect{|
    Hello, world!!
  |}]
```

To reason:

```reason
let%expect_test "foo" = {
  print_endline("Hello, world!");
  %expect
  {|
    Hello, world!!
  |};
};
```

Then I don't get any output when I run `esy dune runtest`. Running the `ocaml` example does give a failing test:

```diff
--- test/example_.ml    2018-12-01 16:34:20.423049283 +0100
+++ test/example_.ml.corrected  2018-12-01 16:37:38.770332016 +0100
@@ -1,5 +1,5 @@
 let%expect_test _ =
   print_endline "Hello, world!!";
   [%expect{|
-    Hello, world!
+    Hello, world!!
   |}]
\ No newline at end of file
```

**Comment by jaredly** [comment]
I assume that this is a problem with ppx_expect, not with reason. I assume that the ppx_expect evaluator only supports `.ml` source files

**Comment by diml** [comment]
Yes, if the extension point syntax is different in reason then ppx_expect would need to be updated to support it. Looking at the example, I guess that extension points in reason don't have `[` and `]`? Although, it's a bit odd that you see no output at all.

**Comment by diml** [comment]
Hmm, I just tried and the corrected file is generated but is named `test.re.ml.corrected` rather than `test.re.corrected`. It's probably the same issue as https://github.com/ocsigen/js_of_ocaml/issues/716 and it should be fixed by ppxlib 0.4.0 that was just released.

**Comment by diml** [comment]
I confirm that it works well with ppxlib 0.4.0
